openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -82,6 +83,12 @@ def content_type_gating_enabled(self):
             course_key=self.course_key,
         )
 
+    @property
+    def can_show_upgrade_sock(self):
+        enrollment = CourseEnrollment.get_enrollment(self.effective_user, self.course_key)
```

Comment:
We query for this as part of check_course_access... I wish there were a way to reuse that. 🤔

Reply:
So there is some request caching going on:

https://github.com/stvstnfrd/edx-platform/blob/fix/courseware-api-verified-mode/common/djangoapps/student/models.py#L1210-L1217